detection-rules/impersonation_sharepoint_body_credential_theft.yml

```diff
@@ -1,16 +1,28 @@
 name: "Brand impersonation: Sharepoint"
 description: |
-  Body contains a Sharepoint logo, a link, and credential theft language.
+  Body, attached images or pdf contains a Sharepoint logo. The message contains a link and credential theft language.
 type: "rule"
 severity: "high"
 source: |
   type.inbound
   and length(body.links) > 0
-  and any(attachments,
-          (.file_type in $file_types_images or .file_type == "pdf")
-          and any(ml.logo_detect(.).brands, .name == "Microsoft SharePoint")
+  and (
+    any(attachments,
+        (.file_type in $file_types_images or .file_type == "pdf")
+        and any(ml.logo_detect(.).brands, .name == "Microsoft SharePoint")
+    )
+    or any(ml.logo_detect(beta.message_screenshot()).brands, .name == "Microsoft SharePoint")
```

Comment:
It won't be every messages just the ones that fail the initial conditions i think, but  I see your point for sure.

Just to play 👿 advocate, we don't gate this function from being used by customers in any capacity, it could easily be used on every message in their environment or cloud if they wanted to. If we're going to release a function and have concerns of it's performance or capabilities at scale, should we release it?

Reply:
> It won't be every messages just the ones that fail the initial conditions i think, but I see your point for sure.

the only initial conditions required here are `type.inbound and length(body.links) > 0`.

I agree in general with that devil's advocate, but I do think there's a difference between what's opt-in and what's opt-out as far as performance impact goes. IIRC, message screenshot isn't typically that bad anyway and doesn't take anywhere nearly as long as LA or Strelka. And we only take screenshots of HTML messages anyway which cuts down on it a bit.

With the amount of rules we run anyway, we're probably running all our functions are large percent of the time anyway

Reply:
In my mind the order of operations was like: 

1. inbound check
2. body links check
3. attachment image/pdf check 
     3a. logo check (assuming 3. true)
     3b. screenshot + logo detect (assuming 3. false)
 4. nlu body thread intent
     4b screenshot + nlu (assuming 4. false)

Reply:
@rw-access word, so are you comfortable with this or should we try to gate it some?

Reply:
ship it! We can always add prevalence later if we need to

Reply:
@morriscode i think what Ross is saying is that 3a and 3b aren't guaranteed to be in that order